pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -0,0 +1,254 @@
+	case shouldDelete:
+		pdb := resourceread.ReadPodDisruptionBudgetV1OrDie(bindata.MustAsset(filepath.Join("pkg/operator/staticpod/controller/guard", "manifests/pdb.yaml")))
+		pdb.ObjectMeta.Name = getGuardPDBName(c.podResourcePrefix)
+		pdb.ObjectMeta.Namespace = c.targetNamespace
+		_, _, err := resourceapply.DeletePodDisruptionBudget(ctx, c.pdbGetter, syncCtx.Recorder(), pdb)
+		if err != nil {
+			klog.Errorf("Unable to delete PodDisruptionBudget: %v", err)
+			errs = append(errs, err)
+		}
+
+		pods, err := c.podLister.Pods(c.targetNamespace).List(labels.SelectorFromSet(labels.Set{"app": "guard"}))
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			for _, pod := range pods {
+				_, _, err = resourceapply.DeletePod(ctx, c.podGetter, syncCtx.Recorder(), pod)
+				if err != nil {
+					klog.Errorf("Unable to delete Pod: %v", err)
+					errs = append(errs, err)
+				}
+			}
+		}
+	case shouldCreate:
+		selector, err := labels.NewRequirement("node-role.kubernetes.io/master", selection.Equals, []string{""})
+		if err != nil {
+			panic(err)
+		}
+		nodes, err := c.nodeLister.List(labels.NewSelector().Add(*selector))
+		if err != nil {
+			return err
+		}
+
+		pods, err := c.podLister.Pods(c.targetNamespace).List(labels.SelectorFromSet(labels.Set{"app": c.podResourcePrefix}))
+		if err != nil {
+			return err
+		}
+
+		klog.V(5).Infof("Rendering guard pdb")
+
+		pdb := resourceread.ReadPodDisruptionBudgetV1OrDie(bindata.MustAsset(filepath.Join("pkg/operator/staticpod/controller/guard", "manifests/pdb.yaml")))
+		pdb.ObjectMeta.Name = getGuardPDBName(c.podResourcePrefix)
+		pdb.ObjectMeta.Namespace = c.targetNamespace
+		if len(nodes) > 1 {
+			minAvailable := intstr.FromInt(len(nodes) - 1)
+			pdb.Spec.MinAvailable = &minAvailable
+		}
+
+		_, _, err = resourceapply.ApplyPodDisruptionBudget(ctx, c.pdbGetter, syncCtx.Recorder(), pdb)
+		if err != nil {
+			klog.Errorf("Unable to apply PodDisruptionBudget changes: %v", err)
+			return err
+		}
+
+		operands := map[string]*corev1.Pod{}
+		for _, pod := range pods {
+			operands[pod.Spec.NodeName] = pod
+		}
+
+		for _, node := range nodes {
+			if _, exists := operands[node.Name]; !exists {
+				klog.Errorf("Missing operand on node %v", node.Name)
+				errs = append(errs, fmt.Errorf("Missing operand on node %v", node.Name))
+				continue
+			}
+
+			if operands[node.Name].Status.PodIP == "" {
+				klog.Errorf("Missing PodIP in operand %v on node %v", operands[node.Name].Name, node.Name)
+				errs = append(errs, fmt.Errorf("Missing PodIP in operand %v on node %v", operands[node.Name].Name, node.Name))
+				continue
+			}
+
+			klog.V(5).Infof("Rendering guard pod for operand %v on node %v", operands[node.Name].Name, node.Name)
+
+			pod := resourceread.ReadPodV1OrDie(bindata.MustAsset(filepath.Join("pkg/operator/staticpod/controller/guard", "manifests/guard-pod.yaml")))
```

Comment:
This seems like a lot of unnecessary reads, why not just read it once when we create the controller and then just update the internal representation as needed and apply.

Reply:
`bindata.MustAsset` already has the internal representation as it is. So it's just a convenient wrapper around it.